modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItLozaTest.java

```diff
@@ -321,4 +301,85 @@ private void validateLastLogEntry(RaftNodeId raftNodeId) {
 
         assertThat(entry, is(notNullValue()));
     }
+
+    @Test
+    void destroysRaftNodeStorages(@InjectConfiguration RaftConfiguration raftConfiguration) throws Exception {
+        ComponentWorkingDir partitionsWorkDir = new ComponentWorkingDir(workDir);
+
+        LogStorageFactory logStorageFactory = SharedLogStorageFactoryUtils.create(
+                clusterService.nodeName(),
+                partitionsWorkDir.raftLogPath()
+        );
+        logStorageFactory = spy(logStorageFactory);
+
+        allComponents.add(logStorageFactory);
+
+        assertThat(logStorageFactory.startAsync(componentContext), willCompleteSuccessfully());
+
+        loza = TestLozaFactory.create(clusterService, raftConfiguration, new HybridClockImpl());
+
+        assertThat(loza.startAsync(componentContext), willCompleteSuccessfully());
+
+        PeersAndLearners configuration = PeersAndLearners.fromConsistentIds(Set.of(clusterService.nodeName()));
+        Peer peer = configuration.peer(clusterService.nodeName());
+
+        RaftGroupListener raftGroupListener = new DrainingRaftGroupListener();
+
+        TestReplicationGroupId replicationGroupId = new TestReplicationGroupId("persistent");
+        var nodeId = new RaftNodeId(replicationGroupId, peer);
+
+        RaftGroupOptionsConfigurer configurer = RaftGroupOptionsConfigHelper.configureProperties(
+                logStorageFactory,
+                partitionsWorkDir.metaPath()
+        );
+
+        CompletableFuture<RaftGroupService> serviceFuture = loza.startRaftGroupNode(
+                nodeId,
+                configuration,
+                raftGroupListener,
+                RaftGroupEventsListener.noopLsnr,
+                null,
+                configurer
+        );
+        assertThat(serviceFuture, willCompleteSuccessfully());
+        RaftGroupService service = serviceFuture.join();
+
+        assertThat(service.run(testWriteCommand("foo")), willCompleteSuccessfully());
+
+        loza.stopRaftNodes(replicationGroupId);
+
+        String groupUri = nodeId.groupId().toString() + "_" + nodeId.peer().consistentId() + "_" + nodeId.peer().idx();
+        Path groupRaftStoragesPath = partitionsWorkDir.metaPath().resolve(groupUri);
+
+        assertTrue(Files.isDirectory(groupRaftStoragesPath));
+
+        loza.destroyRaftNodeStorages(nodeId, configurer);
+
+        verify(logStorageFactory).destroyLogStorage(groupUri);
+        assertFalse(Files.exists(groupRaftStoragesPath));
+    }
+
+    private static class DrainingRaftGroupListener implements RaftGroupListener {
```

Comment:
As a proposal to make this class a separate and package private.

Reply:
It's only used in this class, I think it's better to leave it where it is